src/pages/components/BusArrivalsTimeline.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+    LineChart,
+    Line,
+    XAxis,
+    YAxis,
+    CartesianGrid,
+    Tooltip,
+    Legend,
+    ResponsiveContainer,
+  } from 'recharts'
+
+
+{/* Timeline Graph shows a graph which its X-axis is time or station and Y-axis is time passed (in minutes) */}
+{/* The graph follows the progress of a bus and tracks its correspondence to the schedule */}
+{/* Series 1 indicates the schedule */}
+{/* Series 2 indicates the reality */}
+{/*  */}
+
+
+{/* NEEDS:
+ - timestamp and station data from the schedule
+ - timestamp and station data from the reality
+ - graph
+*/}
+
+const arrayGroup = function <T>(array: T[], f: (item: T) => string) {
+    const groups: Record<string, T[]> = {}
+    array.forEach(function (o) {
+      const group = f(o)
+      groups[group] = groups[group] || []
+      groups[group].push(o)
+    })
+    return Object.keys(groups).map(function (group) {
+      return groups[group]
+    })
+  }
+
+export default function BusArrivalTimeline({
+    data,
+  }: {
+    data: {
+      id: string
+      name: string
+      current: number
+      max: number
+      percent: number
+      gtfs_route_date: string
+      gtfs_route_hour: string
+    }[]
+  })
+  {
+    return (
+        <div className="chart">
+            {arrayGroup(data, (item) => item.id)
+            .filter((group) => group.length > 1 && group.reduce((a, b) => a + b.current, 0) > 1)
+            .map((group) => (
+                <div key={group[0].name}>
+                <h3 className="title">{group[0].name}</h3>
+                <ResponsiveContainer width="100%" height={300}>
+                    <LineChart
+                    data={group
+                        .sort((a, b) => (a.gtfs_route_date > b.gtfs_route_date ? 1 : -1))
+                        .sort((a, b) => (a.gtfs_route_hour > b.gtfs_route_hour ? 1 : -1))}
```

Comment:
I believe it's wrong
consider 1.1.2000 15:00 vs 2.2.2000 14:00

Reply:
this data is irrelevant to the page and was taken from the dashboard page.
looking to update it in the future, I wanted a template to base the new graph